pandas/util/testing.py

```diff
@@ -887,18 +887,16 @@ def assertNotIsInstance(obj, cls, msg=''):
             msg, cls, type(obj)))
 
 
-def assert_categorical_equal(res, exp):
-    assertIsInstance(res, pd.Categorical, '[Categorical] ')
-    assertIsInstance(exp, pd.Categorical, '[Categorical] ')
+def assert_categorical_equal(left, right, obj='Categorical'):
+    assertIsInstance(left, pd.Categorical, '[Categorical] ')
+    assertIsInstance(right, pd.Categorical, '[Categorical] ')
 
-    assert_index_equal(res.categories, exp.categories)
+    assert_index_equal(left.categories, right.categories,
+                       obj='{0}.categories'.format(obj))
+    assert_numpy_array_equal(left.codes, right.codes,
```

Comment:
```
In [2]: tm.assert_numpy_array_equal(np.array([1,2],dtype='int64'),np.array([1,2],dtype='int8'))
Out[2]: True
```
check the dtypes are equal as well (in fact we might want to do that in ``assert_numpy_array_equal``)